app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -923,30 +925,41 @@ class HomeFragment : Fragment() {
             ?.isVisible = tabCount > 0
     }
 
-    private fun displayWallpaperIfEnabled() {
+    private fun observeWallpaperChanges() {
         if (shouldEnableWallpaper()) {
             requireComponents.appStore.flow()
                 .ifChanged { state -> state.wallpaperState.currentWallpaper }
                 .onEach { state ->
-                    // We only want to update the wallpaper when it's different from the default one
-                    // as the default is applied already on xml by default.
-                    when (val currentWallpaper = state.wallpaperState.currentWallpaper) {
-                        Wallpaper.Default -> {
-                            binding.wallpaperImageView.visibility = View.GONE
-                        }
-                        else -> {
-                            val bitmap = requireComponents.useCases.wallpaperUseCases.loadBitmap(currentWallpaper)
-                            bitmap?.let {
-                                it.scaleToBottomOfView(binding.wallpaperImageView)
-                                binding.wallpaperImageView.visibility = View.VISIBLE
-                            }
-                        }
-                    }
+                    showWallpaper(state.wallpaperState.currentWallpaper)
                 }
                 .launchIn(viewLifecycleOwner.lifecycleScope)
         }
     }
 
+    private fun setWallpaperToCurrent() {
+        if (shouldEnableWallpaper()) {
+            val wallpaper = requireComponents.appStore.state.wallpaperState.currentWallpaper
+            runBlockingIncrement {
```

Comment:
Curious why we chose to use `runBlockingIncrement` here instead of launching a coroutine like `lifecycleScope.launch(IO) {}`?

Reply:
Launching it in a `coroutine` is actually what causes the bug. I believe that is because `lifecycleScope`s do not become active until the Fragment is already passed its `onViewCreated` lifecycle state as described in this comment https://github.com/mozilla-mobile/fenix/pull/26515#discussion_r948499212